sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/secrets/DefaultSecretsClient.java

```diff
@@ -23,34 +22,22 @@
  */
 public class DefaultSecretsClient implements SecretsClient {
 
+    private static final String STORE = "default";
     /**
      * URL path prefix for secret store.
      */
-    private static final String STORE_PREFIX = "secret/%s/";
+    private static final String BASE_URL = DcosConstants.SECRETS_BASE_URI + String.format("secret/%s/", STORE);
     private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
 
-    private URL baseUrl;
-    private String store;
     private Executor httpExecutor;
 
-    public DefaultSecretsClient(URL baseUrl, String store, Executor executor) {
-        this.baseUrl = URLUtils.addPathUnchecked(baseUrl, String.format(STORE_PREFIX, store));
-        this.store = store;
-        this.httpExecutor = executor;
-    }
-
-    public DefaultSecretsClient(String store, Executor executor) {
-        this(URLUtils.fromUnchecked(DcosConstants.SECRETS_BASE_URI), store, executor);
-    }
-
     public DefaultSecretsClient(Executor executor) {
-        this("default", executor);
```

Comment:
I see this was the only constructor ever being called.

Are we storing all generated secrets as `secret/default/...`? Should we change that to something like "generated"?

Reply:
Oh, or is a "store" a silo within the secrets api?

Reply:
If it helps to give some context, the complete base URL ends up looking like: `http://master.mesos/secrets/v1/secret/default/`

Reply:
If it helps to give some context, the complete base URL ends up looking like: `http://master.mesos/secrets/v1/secret/default/`

Reply:
The DC/OS has concept of different secrets stores, where default has name `default`. Having a name as a constant makes sense here.

Reply:
Yes, we could extract `default` to a constant (e.g. `DEFAULT_SECRET_STORE_NAME`) to make it clearer what is being set.

Reply:
seconded

Reply:
(Also a comment saying what Martin said would be 👍 )